docker-for-mac/index.md

```diff
@@ -331,14 +334,96 @@ This option removes/resets all Docker data _without_ a reset to factory defaults
 
 ![Uninstall or reset Docker Edge features](images/settings-uninstall-edge.png)
 
+## Adding security certificates
+
+You can add server side (registry) and client side certificates to verify the identity of these entities.
+
+### Adding custom CA certificates (server side)
+
+All trusted certificate authorities (CAs) (root or intermediate) are supported.
+Docker for Mac creates a certificate bundle of all user-trusted CAs based on the
+Mac Keychain, and appends it to Moby trusted certificates. So if an enterprise
+SSL certificate is trusted by the user on the host, it will be trusted by Docker
+for Mac.
+
+To manually add a custom, self-signed certificate, start by adding
+the certificate to the Mac’s keychain, which will be picked up by Docker for
+Mac. Here is an example.
+
+```bash
+sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain ca.crt
+```
+
+Or, if you prefer to add the certificate to your own local keychain only (rather
+than for all users), run this command instead:
+
+```
+security add-trusted-cert -d -r trustRoot -k ~/Library/Keychains/login.keychain ca.crt
+```
+
+See also, [Directory structures for
+certificates](#directory-structures-for-certificates).
+
+> **Note:** You need to restart Docker for Mac after making any changes to
+the keychain or to the `~/.docker/certs.d` directory in order for
+the changes to take effect.
+
+For a complete explanation of how to do this, see the blog post [Adding
+Self-signed Registry Certs to Docker & Docker for
+Mac](http://container-solutions.com/adding-self-signed-registry-certs-docker-mac/).
+
+### Adding client certificates
+
+Starting with Docker for Mac 17.06.0-ce, you do not have to push your
+certificates with `git` commands anymore. You can put your client certificates
+in `~/.docker/certs.d/<MyRegistry>:<Port>/client.cert` and
+`~/.docker/certs.d/<MyRegistry>:<Port>/client.key`.
+
+When the Docker for Mac application starts up, it copies certificates from your
+Mac folders to `~/.docker/certs.d` in the database.
+
+> * You need to restart Docker for Mac after making any changes to
+ the keychain or to the `~/.docker/certs.d` directory in order for
+ the changes to take effect.
+>
+> * The registry cannot be listed as an _insecure registry_ (see [Docker
+Daemon](/docker-for-mac/index.md#docker-daemon)). Docker for Mac will ignore
+certificates listed under insecure registries, and will not send client
+certificates. Commands like `docker run` that attempt to pull from
+the registry will produce error messages on the command line, as well as on the
+registry.
+
+### Directory structures for certificates
+
+If you have this directory structure, you do not need to manually add the CA
+certificate to your Mac OS system login:
+
+```
+/Users/<user>/.docker/certs.d/
+└── 192.168.203.139:5858
```

Comment:
Since `~/docker/certs.d/<MyRegistry>:<Port>/client.key` etc. was mentioned above, for consistency do we want to change the folder name from `192.168.203.139:5858` to `<MyRegistry>:<Port>`? 

Alternately, since https://docs.docker.com/engine/security/certificates/#creating-the-client-certificates uses "localhost:5000" as the hostname with port, maybe we should use that instead?  Similar with the example below.